### ISSUE TITLE
Status bar color not getting changed

### ISSUE THREAD

**Issue by ask007java** [react-native-community/react-native-navbar#221]
Am I doing something wrong? the statusbar color is not changing. My code below
```javascript
 <NavigationBar
                  title={{title:'Notification', tintColor:'white'}}
                  tintColor="#ef5350"
                  statusBar={{style: 'light-content',tintColor:'#ef5350'}}
                  leftButton={leftButton}
                  rightButton={rightButton}
              />
```

<img width="308" alt="screen shot 2017-04-18 at 10 18 08 pm" src="https://cloud.githubusercontent.com/assets/6759476/25162010/0692c35c-2485-11e7-9a22-bf0db273d35a.png">

**Comment by jenskuhrjorgensen** [comment]
I'm having the same problem! :'(

According to the react-native Statusbar docs it is possible to change the style (and background color) of the status bar:
```
   <StatusBar
     backgroundColor="blue"
     barStyle="light-content"
   />
```

But in react-native-navbar it seems that only the iOS status bar is being styled:
  ```
customizeStatusBar() {
    const { statusBar } = this.props;
    if (Platform.OS === 'ios') {
      if (statusBar.style) {
        StatusBar.setBarStyle(statusBar.style);
      }

      const animation = statusBar.hidden ?
        statusBar.hideAnimation : statusBar.showAnimation;

      StatusBar.showHideTransition = animation;
      StatusBar.hidden = statusBar.hidden;
    }
  }
```

I hope that the Android status bar styling will be added soon :)

**Comment by alexindigo** [comment]
You can use just regular StatusBar component – https://facebook.github.io/react-native/docs/statusbar.html
It has backgroundColor property.

**Comment by jenskuhrjorgensen** [comment]
But can you use that in conjunction with react-native-navbar?

**Comment by alexindigo** [comment]
Yep. It doesn't render anything by itself, just modifies settings (I know feels counter-intutive, but it's what we get for using react) :)

Here is example: https://github.com/alexindigo/ndash/blob/master/app/components/Chrome.js#L151
_(the app uses `react-native-navbar`)_

**Comment by jenskuhrjorgensen** [comment]
Nice, thanks! :D
However, I still think that styling the Android status bar through react-native-navbar should be possible.

**Comment by alexindigo** [comment]
For sure.

**Comment by JaxGit** [comment]
Man... 
Why are we getting triggered by nextProps but keep setting with old props?...

 `
  componentDidMount() {
    this.customizeStatusBar();
  }

  componentWillReceiveProps() {
    this.customizeStatusBar();
  }

  customizeStatusBar() {
    const { statusBar } = this.props;
    if (Platform.OS === 'ios') {
      if (statusBar.style) {
        StatusBar.setBarStyle(statusBar.style);
      }
`

shouldn't these be:
`
  componentDidMount() {
    this.customizeStatusBar(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.customizeStatusBar(nextProps);
  }

  customizeStatusBar(anyProps) {
    const { statusBar } = anyProps;
    if (Platform.OS === 'ios') {
      if (statusBar.style) {
        StatusBar.setBarStyle(statusBar.style);
      }
`

**Comment by Kingston7** [comment]
The statusbar color is not changing,i am using iphone

**Comment by jinhale** [comment]
None of the advice on [React Native](https://facebook.github.io/react-native/docs/statusbar.html) worked for me. I ended up using the advice of [React Navigation](https://reactnavigation.org/docs/en/headers.html) for my Android app.

**Comment by mujibjan** [comment]
my status bar color is also not changing here is the code 
"expo": {

    "androidStatusBar": {
      "backgroundColor": "#18c22a"
    },

**Comment by mujibjan** [comment]
> If you are using Expo for React Native then here is the solution for setting Android Status Bar Color.
> First of all, In your app.json file add the code:
> 
> ```
> {
>   "expo": {
>     "sdkVersion": "Your given Version",
>     "androidStatusBar": {
>        "backgroundColor": "#4e2ba1" (Your desirable android Status Bar Color before the app loads)
>     }
>    }    
> }
> ```
> And then Go to Your Main Component or App.js, import 'StatusBar' from 'react-native'. Then add Following Code in return:
> 
> ```
> return(
>    <View style={{flex: 1}}>  (Do not forget to style flex as 1)
>       <StatusBar translucent backgroundColor="rgba(0,0,0,0.2)"/>
>       <Your Code>
>    </View>
> );
> ```
> Here, we are setting the status bar color as Black but with 0.2 opacity. Your statusBar Color will be the same as your headerStyle background Color for Stack Navigator but a bit darker. For standard Android App, this is how the StatusBar color is set. Also, Make it translucent so that your app draws under the status Bar and looks nice.
> 
> It hope this works perfectly for you.
This is not working i have done such this.

"expo": {
    "name": "Talem Islam Radio",
    "androidStatusBar": {
      "backgroundColor": "#18c22a"
    },

**Comment by atemp21** [comment]
workaround by adding a view under the status bar as document in Expo
`import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Constants } from 'expo';

const styles = StyleSheet.create({
  statusBar: {
    backgroundColor: "#C2185B",
    height: Constants.statusBarHeight,
  },

  // rest of the styles
});

const MyComponent = () => {
  <View>
    <View style={styles.statusBar} />
    {/* rest of the content */}
  </View>
}`

https://docs.expo.io/versions/latest/guides/configuring-statusbar/

**Comment by dkocich** [comment]
in my IgniteCli project, it started working with a default barStyle
`<StatusBar barStyle='default' backgroundColor={Colors.blue} />`

**Comment by ayoubamine** [comment]
[react-native-general-statusbar][1]

npm i --save react-native-general-statusbar

```javascript
import GeneralStatusBarColor from "react-native-general-statusbar";
 
function MyComponent() {
    return (
        <View>
            <GeneralStatusBarColor
                backgroundColor="#00AEEF"
                barStyle="light-content"
            />
        </View>
    );
}
```

  [1]: https://www.npmjs.com/package/react-native-general-statusbar

**Comment by Alain00** [comment]
I've the same situation, i'm testing my application on ExpoApp, 

> 
> 
> If you are using Expo for React Native then here is the solution for setting Android Status Bar Color.
> First of all, In your app.json file add the code:
> 
> ```
> {
>   "expo": {
>     "sdkVersion": "Your given Version",
>     "androidStatusBar": {
>        "backgroundColor": "#4e2ba1" (Your desirable android Status Bar Color before the app loads)
>     }
>    }    
> }
> ```
> 
> And then Go to Your Main Component or App.js, import 'StatusBar' from 'react-native'. Then add Following Code in return:
> 
> ```
> return(
>    <View style={{flex: 1}}>  (Do not forget to style flex as 1)
>       <StatusBar translucent backgroundColor="rgba(0,0,0,0.2)"/>
>       <Your Code>
>    </View>
> );
> ```
> 
> Here, we are setting the status bar color as Black but with 0.2 opacity. Your statusBar Color will be the same as your headerStyle background Color for Stack Navigator but a bit darker. For standard Android App, this is how the StatusBar color is set. Also, Make it translucent so that your app draws under the status Bar and looks nice.
> 
> It hope this works perfectly for you.

I'm having the same situation. And this thoes't work for me, i'm only able to hide the StatusBar, but unable to see the changes on any other propertie

**Comment by cogell** [comment]
@ayoubamine this didnt seem to work for me on Android 10
